redash/query_runner/impala_ds.py

```diff
@@ -47,8 +47,12 @@ def configuration_schema(cls):
                     "type": "number"
                 },
                 "protocol": {
-                    "type": "string",
-                    "title": "Please specify beeswax or hiveserver2"
+                    "type": "enum",
+                    "options": [
+                        {"value": "beeswax"},
+                        {"value": "hiveserver2"}
+                    ],
```

Comment:
This won't pass schema validation. A valid enum is:

```json
{
    "protocol": {
        "type": "string",
        "enum": ["beeswax", "hiveserver2"]
    }
}
```

We might want to add our own extension to provide friendly names for options:

```
{
    "protocol": {
        "type": "string",
        "extendedEnum": [{"value", "beeswax", "name": "Beeswax"}, {"value": "hiveserver2", "name": "Hive Server 2"}]
    }
}
```

And we will transform `extendedEnum` into `enum` before passing it into `jsonschema.validate` call.

Reply:
>We might want to add our own extension to provide friendly names for options

That's exactly why I went with the list of objects on that, but yeah, it does make sense to permit a simplified version.

> And we will transform extendedEnum into enum before passing it into jsonschema.validate call.

That's the only thing missing in terms of backend code?

Reply:
> That's the only thing missing in terms of backend code?

I think so.